### PR TITLE
Init docs for AWS IAM Roles regcreds

### PR DESCRIPTION
**Pull request by maroshii**
We still need to add the full steps for the OIDC Federation section but opening early to get feedback on structure and whatnot

### CODE REVIEW

**Comment by codyjlandstrom** [comment]
Structure looks good 👍

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 1]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
 - **Hostname**: the default registry endpoint is `https://{AWS_ACCOUNT_ID}.dkr.ecr.{REGION}.amazonaws.com`
 - **Username**: `AccessKeyId` from the previous step
 - **Password**: `SecretAccessKey` from the previous step
+
+# Using OIDC Federation
+
+## Step 1: Create the Identity Provider
+
+```bash
+OIDC_ENDPOINT=https://container.googleapis.com/v1/projects/my-project-12345/locations/us-central1/clusters/development
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:72]: are we showing this endpoint in the Okteto UI? If yes, let's document how to get it from there

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 2]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+AUDIENCE=registry.okteto.dev
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:73]: is this an arbitrary string?

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 3]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+EOT
+
+aws iam create-role --role-name my-private-registry --assume-role-policy-document file://trust-policy.json
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "my-private-registry",
+        "RoleId": "AR...",
+        "Arn": "arn:aws:iam::112233445566:role/my-private-registry",
+        "CreateDate": "2024-06-10T15:04:05+00:00",
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:108]: do we need to set this field?

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 4]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+  ]
+}
+EOT
+
+aws iam create-role --role-name my-private-registry --assume-role-policy-document file://trust-policy.json
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "my-private-registry",
+        "RoleId": "AR...",
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:106]: where is this value coming from?

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 5]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+
+If you only need read access you can use `AmazonEC2ContainerRegistryReadOnly` instead.
+
+
+## Step 3: Configure the credentials in Okteto
+
+Add the following registry credentials to the [Admin Registry Credentials view](index.mdx#add-registry-credentials):
+
+- **Type**: `AWS IAM Role`
+- **Hostname**: the default registry endpoint is `https://{AWS_ACCOUNT_ID}.dkr.ecr.{REGION}.amazonaws.com`
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:139]:
```suggestion
- **Hostname**: The ECR registry endpoint is `https://{AWS_ACCOUNT_ID}.dkr.ecr.{REGION}.amazonaws.com`
```

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 6]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+- **Role ARN**: The Role ARN from the previous step
```
Comment (pchico83) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:140]: Can we also refer to the value in our sample?

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 7]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+    }
+}
+```
+
+Attach an EC2 Container Registry policy that allows to pull and push the the registry:
```
Comment (codyjlandstrom) [inline src/content/admin/registry-credentials/amazon-ecr.mdx:125]:
```suggestion
Attach an EC2 Container Registry policy that allows you to pull and push from the registry:
```

File: src/content/admin/registry-credentials/amazon-ecr.mdx (hunk 2 above)
Comment (pchico83) [reply src/content/admin/registry-credentials/amazon-ecr.mdx:73]: @maroshii let's explain that as part of this doc

File: src/content/admin/registry-credentials/amazon-ecr.mdx (hunk 1 above)
Comment (pchico83) [reply src/content/admin/registry-credentials/amazon-ecr.mdx:72]: I think frontend is done already and available in staging

File: src/content/admin/registry-credentials/amazon-ecr.mdx [hunk 8]
```diff
@@ -55,7 +59,83 @@ Remember the value of `AccessKeyId` and `SecretAccessKey`. You will need them in
...
+
+
+## Step 3: Configure the credentials in Okteto
+
+Add the following registry credentials to the [Admin Registry Credentials view](index.mdx#add-registry-credentials):
+
+- **Type**: `AWS IAM Role`
+- **Hostname**: the default registry endpoint is `https://{AWS_ACCOUNT_ID}.dkr.ecr.{REGION}.amazonaws.com`
+- **Role ARN**: The Role ARN from the previous step
+- **Audience**: The Audience used for the Identity Provider
```
[the remark this answers is not in the document]
Comment (pchico83) [reply src/content/admin/registry-credentials/amazon-ecr.mdx:164]: In the sample, seems like we are using [registry.okteto.dev](https://github.com/okteto/docs/pull/753/files#diff-3e3b33ce4bb970fa818d1ed2115646a9bedd6442771d6e9db7b843dbd05d7eb6R73). Referring to the sample helps to understand the meaning of this field

**Comment by bgonp** [comment]
I took the liberty of updating the screenshots with the last version of the UI and also added a missing `import` so the preview is now successfully deployed, feel free to [take a look at it](https://deploy-preview-753--okteto-docs.netlify.app/docs/1.22/admin/registry-credentials/)

(I also added a tiny comment in the readme about naming version-specific images as [stated here](https://github.com/okteto/docs/pull/572#issue-2058386676))

**Pull request merged**